ext/node/ops/http.rs

```diff
@@ -133,89 +213,33 @@ where
   }
 
-  let cancel_handle = CancelHandle::new_rc();
-  let cancel_handle_ = cancel_handle.clone();
-
-  let fut = async move {
-    client
-      .send(request)
-      .map_err(Into::into)
-      .or_cancel(cancel_handle_)
-      .await
-  };
-
-  let request_rid = state.resource_table.add(FetchRequestResource {
-    future: Box::pin(fut),
-    url,
-  });
-
-  let cancel_handle_rid =
-    state.resource_table.add(FetchCancelHandle(cancel_handle));
-
-  Ok(FetchReturn {
-    request_rid,
-    cancel_handle_rid: Some(cancel_handle_rid),
-  })
-}
+  let res = sender.send_request(request).map_err(Error::from).boxed();
+  let rid = state
+    .borrow_mut()
+    .resource_table
+    .add(NodeHttpClientResponse {
+      response: res,
+      url: url.clone(),
+    });
 
-#[derive(Default, Serialize)]
-#[serde(rename_all = "camelCase")]
-pub struct NodeHttpFetchResponse {
-  pub status: u16,
-  pub status_text: String,
-  pub headers: Vec<(ByteString, ByteString)>,
-  pub url: String,
-  pub response_rid: ResourceId,
-  pub content_length: Option<u64>,
-  pub remote_addr_ip: Option<String>,
-  pub remote_addr_port: Option<u16>,
-  pub error: Option<String>,
+  Ok(rid)
 }
 
 #[op2(async)]
 #[serde]
-pub async fn op_node_http_fetch_send(
+pub async fn op_node_http_await_response(
   state: Rc<RefCell<OpState>>,
   #[smi] rid: ResourceId,
-) -> Result<NodeHttpFetchResponse, FetchError> {
-  let request = state
+) -> Result<NodeHttpResponse, AnyError> {
+  let resource = state
     .borrow_mut()
     .resource_table
-    .take::<FetchRequestResource>(rid)
-    .map_err(FetchError::Resource)?;
-
-  let request = Rc::try_unwrap(request)
-    .ok()
-    .expect("multiple op_node_http_fetch_send ongoing");
-
-  let res = match request.future.await {
-    Ok(Ok(res)) => res,
-    Ok(Err(err)) => {
-      // We're going to try and rescue the error cause from a stream and return it from this fetch.
-      // If any error in the chain is a hyper body error, return that as a special result we can use to
-      // reconstruct an error chain (eg: `new TypeError(..., { cause: new Error(...) })`).
-      // TODO(mmastrac): it would be a lot easier if we just passed a v8::Global through here instead
-
-      if let FetchError::ClientSend(err_src) = &err {
-        if let Some(client_err) = std::error::Error::source(&err_src.source) {
-          if let Some(err_src) = client_err.downcast_ref::<hyper::Error>() {
-            if let Some(err_src) = std::error::Error::source(err_src) {
-              return Ok(NodeHttpFetchResponse {
-                error: Some(err_src.to_string()),
-                ..Default::default()
-              });
-            }
-          }
-        }
-      }
-
-      return Err(err);
-    }
-    Err(_) => return Err(FetchError::RequestCanceled),
-  };
+    .take::<NodeHttpClientResponse>(rid)?;
+  let resource = Rc::try_unwrap(resource)
+    .map_err(|_| bad_resource("NodeHttpClientResponse"))?;
 
+  let res = resource.response.await?;
```

Comment:
I think we should remap the error here from a generic hyper error to a concrete type we handle in JS